hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -222,24 +220,144 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
+      Iterable<List<String>> tableNameSets = Iterables.partition(tableNames, 100);
+
+      for (List<String> tableNameSet : tableNameSets) {
+        filteredTableIdentifiers.addAll(
+            filterIcebergEntities(
+                tableNameSet, namespace, HiveOperationsBase.ICEBERG_VIEW_TYPE_VALUE));
+      }
+
+      return filteredTableIdentifiers;
+    } catch (UnknownDBException e) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all views under namespace " + namespace, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listViews", e);
+    }
+  }
+
+  private List<TableIdentifier> filterIcebergEntities(
+      List<String> tableNames, Namespace namespace, String tableTypeProp)
+      throws TException, InterruptedException {
+    List<Table> tableObjects =
+        clients.run(client -> client.getTableObjectsByName(namespace.level(0), tableNames));
+    return tableObjects.stream()
+        .filter(
+            table ->
+                table.getParameters() != null
+                    && tableTypeProp.equalsIgnoreCase(
+                        table.getParameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP)))
+        .map(table -> TableIdentifier.of(namespace, table.getTableName()))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  @SuppressWarnings("FormatStringAnnotation")
+  public void renameView(TableIdentifier from, TableIdentifier to) {
     if (!namespaceExists(to.namespace())) {
       throw new NoSuchNamespaceException(
           "Cannot rename %s to %s. Namespace does not exist: %s", from, to, to.namespace());
     }
+    renameContent(from, to, HiveOperationsBase.ContentType.VIEW);
+  }
+
+  private void renameContent(
+      TableIdentifier fromIdentifier,
+      TableIdentifier toIdentifier,
+      HiveOperationsBase.ContentType contentType) {
+    if (!isValidIdentifier(fromIdentifier)) {
+      throw new NoSuchViewException("Invalid identifier: %s", fromIdentifier);
+    }
+
+    TableIdentifier to = removeCatalogName(toIdentifier);
+    Preconditions.checkArgument(isValidIdentifier(to), "Invalid identifier: %s", to);
+    if (!namespaceExists(to.namespace())) {
+      throw new NoSuchNamespaceException(
+          "Cannot rename %s to %s. Namespace does not exist: %s",
+          fromIdentifier, to, to.namespace());
+    }
 
     String toDatabase = to.namespace().level(0);
-    String fromDatabase = from.namespace().level(0);
-    String fromName = from.name();
+    String fromDatabase = fromIdentifier.namespace().level(0);
+    String fromName = fromIdentifier.name();
 
     try {
       Table table = clients.run(client -> client.getTable(fromDatabase, fromName));
-      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, from));
+      switch (contentType) {
```

Comment:
what do you think to pushdown the switch to a method in HiveOperationsBase called validateTable()?  (pass in content).  The switch statement there should choose the right string for the error message.